src/dotnet-test-xunit/CommandLine.cs

```diff
@@ -59,13 +60,24 @@ protected CommandLine(IReadOnlyList<IRunnerReporter> reporters, string[] args, P
         static XunitProject GetProjectFile(List<Tuple<string, string>> assemblies)
         {
             var result = new XunitProject();
-
+#if NET451
+            // When running on desktop, use the executable from the publish directory rather than the output directory. This is required
+            // for xunit to locate the dependency closure.
+            var publishPath = PlatformServices.Default.Application.ApplicationBasePath;
```

Comment:
@pakrym suggested changing dotnet-test command to pass in the `RuntimeFiles` path rather than the `CompilationFiles` path. This would avoid this hack.